c2rust-analyze/src/util.rs

```diff
@@ -89,8 +89,21 @@ pub enum Callee<'tcx> {
         mutbl: Mutability,
     },
 
-    /// A built-in or standard library function that requires no special handling.
-    MiscBuiltin,
+    /// A [`Trivial`] library function that requires no special handling.
+    ///
+    /// A function is [`Trivial`] if it has no argument or return types that are or contain a pointer.
+    /// Note that "contains a pointer" is calculated recursively.
+    /// There must not be any raw pointer accessible from that type.
+    ///
+    /// Int-to-ptr casts a la [`std::ptr::from_exposed_addr`] are not considered for this,
+    /// as doing so is very difficult and out of scope for now.
```

Comment:
```suggestion
    /// We ignore the possibility that a function taking `usize` might perform int-to-ptr casts internally,
    /// as handling such casts is very difficult and out of scope for now.
```

Reply:
In fd48ed87242a603bced61135a612d57c182b9531, I made this change except for the "taking `usize`" part, as it doesn't have to be a `usize`, it could be any integer, or it could just be fabricated from nothing.